--- ./src/apk_adb.c	2024-11-13 07:22:28.096000000 +0000
+++ ./src/apk_adb.c	2024-11-13 07:19:34.292000000 +0000
@@ -176,7 +176,7 @@
 
 static adb_val_t version_fromstring(struct adb *db, apk_blob_t val)
 {
-	if (!apk_version_validate(val)) return ADB_ERROR(APKE_PKGVERSION_FORMAT);
+	// if (!apk_version_validate(val)) return ADB_ERROR(APKE_PKGVERSION_FORMAT);
 	return adb_w_blob(db, val);
 }
 
@@ -336,8 +336,7 @@
 	int op;
 
 	if (apk_dep_parse(bdep, &bname, &op, &bver) != 0) goto fail;
-	if ((op & APK_DEPMASK_CHECKSUM) != APK_DEPMASK_CHECKSUM &&
-	    !apk_version_validate(bver)) goto fail;
+	if ((op & APK_DEPMASK_CHECKSUM) != APK_DEPMASK_CHECKSUM) goto fail;
 
 	if (apk_blob_spn(bname, APK_CTYPE_DEPENDENCY_NAME, NULL, NULL)) goto fail;
 
